tests/integration/run_integration_tests.sh

```diff
@@ -0,0 +1,222 @@
+cleanup_on_sigint() {
+    if [[ ${IS_INTEGRATION_TEST_RUNNING} == "true" ]]; then
+      preserve_info_and_cleanup
+    fi
+}
+
+trap "cleanup_on_sigint" SIGINT
+trap "cleanup" SIGTERM EXIT
+
+help()
+{
+   # Display Help
+   echo "This script will run the integration test suite locally. By default, all integration tests besides of the 'self-hosted-notary' test will be run (addable with -e (extended) flag)."
+   echo
+   echo "Flag   Values                       Meaning         Description"
+   echo "s      space-separated list         skip            Provided integration tests will be skipped. Cannot be used with -r (run-only) flag."
+   echo "       of integration test names"
+   echo "       as stings"
+   echo "r      space-separated list         run-only        Only the provided integrations tests will be run. Cannot be used with -s (skip) flag."
+   echo "       of integration test names"
+   echo "       as stings"
+   echo "e                                   extended        Include 'self-hosted-notary' test."
+   echo "c       <'minikube'|'kind'>         cluster         Specifies the cluster that should be used for integration testing. Must be either 'minikube' or 'kind'. (Default: 'minikube')"
+   echo "p       <path>                      preserve-logs   Must be a relative or absolute path. If set, log files of failed tests are copied that path."
+   echo "h                                   help            Display this help."
+   echo
+   exit 0
+}
+
+while getopts 'hs:r:c:d:p:e' FLAG; do
+  case "$FLAG" in
+    h)
+      help
+    ;;
+    s)
+      if [[ "${INTEGRATION_TESTS_TO_RUN:-"not-set"}" == "not-set" ]]; then
+        readarray -d ' ' -t INTEGRATION_TESTS_TO_SKIP <<< "${OPTARG}"
+      else
+        echo "Aborting: Please specify either the run-only (-r) or the skip (-s) flag, not both."
+      fi
+    ;;
+    r)
+      if [[ "${INTEGRATION_TESTS_TO_SKIP:-"not-set"}" == "not-set" ]]; then
+      readarray -d ' ' -t INTEGRATION_TESTS_TO_RUN <<< "${OPTARG}"
+      else
+        echo "Aborting: Please specify either the run-only (-r) or the skip (-s) flag, not both."
+        exit 1
+      fi
+    ;;
+    c)
+      CLUSTER="${OPTARG}"
+      if ! [[ "${CLUSTER}" =~ ^(minikube|kind)$ ]]; then
+        echo "Aborting: This script only supports local integration tests with either minikube or kind."
+        exit 1
+      fi
+    ;;
+    p)
+      DESTINATION="${OPTARG}"
+    ;;
+    e)
+      EXTENDED="true"
+  esac
+done
+shift
+
+mkdir tmp
+
+export COSIGN_PUBLIC_KEY="-----BEGIN PUBLIC KEY-----<br>MFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAEQ70fJ/NA109p3/43cTgwnLQ+HsvK<br>jFK/0L1kLUQO+e2cL+3+y2s4IykwTxfSnqOSA0kEy1rTrhrXLCox4uRmSQ==<br>-----END PUBLIC KEY-----"
+
+if [[ "${CLUSTER:-"minikube"}" == "minikube" ]]; then
+  if [[ "$(docker inspect minikube > /dev/null 2>&1 | jq -r .[].State.Status || echo 'container not found')" != "running" ]]; then
+    echo "Minikube cluster configured. You need a running minikube cluster. Exiting..."
```

Comment:
Is there a 'no(t)' missing?

Reply:
No that's to tell user that what cluster is actually configured. Especially if they hav a kind cluster running, but forgot to provide -c flag, the default here is minikube and the message will be `Minkube configured`. Should I reword that somehow?

Reply:
Ah, that's "You configured to run the tests on Minikube, but there's no running Minikube." I read "You have a configured Minikube cluster, but you need a running one". My head would be glad if there were a rewording 😁